vortex-file/src/segments/cache.rs

```diff
@@ -31,34 +31,35 @@ impl SegmentCache for NoOpSegmentCache {
     }
 }
 
-#[derive(Default)]
-pub(crate) struct InMemorySegmentCache(RwLock<HashMap<SegmentId, ByteBuffer>>);
+pub(crate) struct InMemorySegmentCache(Cache<SegmentId, ByteBuffer>);
+
+impl InMemorySegmentCache {
+    pub fn new(
+        builder: CacheBuilder<SegmentId, ByteBuffer, Cache<SegmentId, ByteBuffer, RandomState>>,
+    ) -> Self {
+        Self(
+            builder
+                .weigher(|_, buffer| {
```

Comment:
nit - I think we probably want to set the [eviction policy](https://docs.rs/moka/latest/moka/policy/struct.EvictionPolicy.html) to `lru`

Reply:
Why not LFU?

Reply:
it won't (necessarily) cache recent segments, which is probably the behavior you always want for operations the re-read the same column like evaluating a filter and then returning it.